src/libstd/sync/barrier.rs

```diff
@@ -141,7 +141,7 @@ impl Barrier {
     pub fn wait(&self) -> BarrierWaitResult {
         let mut lock = self.lock.lock().unwrap();
         let local_gen = lock.generation_id;
-        lock.count += 1;
+        lock.count.wrapping_add(1);
```

Comment:
How can this ever overflow? Wouldn't that require more than `usize::MAX_VALUE` threads?